rfcs/051-concepts-adapters/README.md

```diff
@@ -0,0 +1,60 @@
+
+![A more detailed view of how a concepts ingestor might work](https://user-images.githubusercontent.com/4429247/177571629-56f7dd7b-9c43-4f38-bb11-aae53ed0c2e6.png)
+
+The "transform" step here is (a) the most ill-defined and (b) the only step that needs to differ for LCSH and LCNAF ingestors.
+
+### Transforming external concepts 
+
+[RFC 050](https://github.com/wellcomecollection/docs/tree/main/rfcs/050-concepts-api) outlined the rough design of a concepts API: obviously, we need to store the external concepts in a way that is sufficient to construct these API documents. We also don't want to extraneous data, as the majority of the concepts won't be used in the catalogue. 
+
+Based on this, we might transform the LCSH documents to something like:
+
+```json
+{
+  "id": "lc-subjects/sh12345678", // QName-type identifier 
+  "identifier": {
+    "value": "sh12345678",
+    "identifierType": "lc-subjects",
+    "type": "Identifier"
+  },
+  "label": "<value of skos:prefLabel>",
+  "alternativeLabels": [
+    "<values of skos:altLabel>"
+  ]
+}
+
+```
+### Ingesting changes 
+
+_This section is a bit speculative - we don't necessarily need the answers right now._
+
+We can start by just writing all of the source into an empty index - this is inefficient and we won't know what (if anything) changed, but it will never be wrong.
+
+To know what's changed, we have to start comparing to what's there already. This is of course very inefficient, but we can pass that inefficiency onto the ES cluster by using the [`doc_as_upsert`](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update.html#doc_as_upsert) feature of the `update` API alongside the [`noop`](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update.html#_detect_noop_updates) feature: with a single API call we'll be able to know whether anything was added.
+
+This leaves us with detecting deletions: the least likely kind of change. I don't think this needs to be addressed right now - if it's something we become aware of we can just write into an empty index again.
+
+## Questions
+
+- The format proposed above doesn't follow the existing schema for `identifierType` - rather than just the id we have previously used an object for `identifierType`, like
+  ```json
+  {
+    "id": "lc-subjects",
+    "label": "Library of Congress Subject Headings (LCSH)",
+    "type": "IdentifierType"
+  }
+  ```
+  Is this OK? **Yes, we prefer this**
+- Is the unused `id` on the documents actually necessary? **It's useful to be able to construct queries for these concepts, and the identifier proposed is scalable to various source schemas**
```

Comment:
I did wonder if we need the `id` on the document itself; rather than as the `_id` of the Elastic document.

Presumably if we wanted to query by ID, we'd query on the structured `identifier` object instead?

I can think of scenarios where it's useful, and it does no harm, so I think it's fine.

Reply:
I did actually mean this to be the `_id` so will correct that